packages/contracts/contracts/factory/RepoFactory.sol

```diff
@@ -0,0 +1,92 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity 0.8.10;
+
+import "../utils/Proxy.sol";
+import "../registry/APMRegistry.sol";
+import "../APM/Repo.sol";
+
+/// @title RepoFactory to create a Repo
+/// @author Sarkawt Noori - Aragon Association - 2022
+/// @notice This contract is used to create a Repo and register it on APMRegistry contract.
+contract RepoFactory {
+    APMRegistry public apmRegistry;
+    address public repoBase;
+
+    error ApmRegEmpityName();
+
+    constructor(APMRegistry _apmRegistry) {
+        apmRegistry = _apmRegistry;
+
+        setupBases();
+    }
+
+    /// @notice Create new repo in registry with `_name`
+    /// @param _name Repo name, must be ununsed
+    /// TODO: Rethink if it need permission to prevent it from getting poluted, same for newRepoWithVersion
```

Comment:
yes,
but WDYT?
- should we resolve this in the scope of this task, or in another task? because it will raise the question of who will be the permission granter?